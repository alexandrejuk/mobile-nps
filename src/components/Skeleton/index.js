import React from 'react'
import PropTypes from 'prop-types'

import styles from './style.module.css'

const Skeleton = ({
  borderRadius,
  height,
  margin,
  width,
}) => (
  <div
    className={styles.skeleton}
    style={{
      borderRadius,
      height,
      margin,
      width,
    }}
  />
)

Skeleton.propTypes = {
  borderRadius: PropTypes.string,
  height: PropTypes.string,
  margin: PropTypes.string,
  width: PropTypes.string,
}

Skeleton.defaultProps = {
  borderRadius: '3px',
  height: '20px',
  margin: '4px',
  width: '110px',
}

export default Skeleton
