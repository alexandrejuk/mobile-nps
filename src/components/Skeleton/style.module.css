.skeleton {
  border-radius: 3px;
  background-image: linear-gradient(90deg, #9e9e9e 0%, #f5f5f5 100%)
}
