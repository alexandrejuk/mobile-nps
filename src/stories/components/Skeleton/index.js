import React from 'react'
import { Skeleton } from '../../../components'

export default {
  title: 'Components/Skeleton',
}

export const Default = () => (
  <Skeleton />
)
