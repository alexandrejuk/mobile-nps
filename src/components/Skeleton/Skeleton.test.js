import React from 'react'
import {
  render,
} from '@testing-library/react'

import Skeleton from '.'

describe('Skeleton component', () => {
  it('should render correctly primary button', () => {
    const { container } = render(
      <Skeleton />
    )
    expect(container).toMatchSnapshot()
  })
})
