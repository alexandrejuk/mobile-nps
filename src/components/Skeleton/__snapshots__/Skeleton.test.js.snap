// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Skeleton component should render correctly primary button 1`] = `
<div>
  <div
    class="skeleton"
    style="border-radius: 3px; height: 20px; margin: 4px; width: 110px;"
  />
</div>
`;
